### PR TITLE
Performance and stability improvements

### PR DESCRIPTION
**Pull request by lkrocek**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # BrightWebSocket
 
-A ScreneGraph websocket client library written in BrightScript
+A ScreneGraph websocket client library written in BrightScript. It is written to work in separate task to don't affect main thread of application.
```
Comment (lukaslohnicky) [inline readme.md]: > to don't affect

to not affect

File: readme.md [hunk 2]
```diff
@@ -44,6 +46,23 @@ end function
 
 For a working sample app see the "test" folder. Its contents can be zipped for
  installation as a dev channel on a Roku.
+ 
+# Additional information
+
+## A differences between `roUrlTransfer` and `roStreamSocket`
+
+The `roUrltransfer` is sending bigger data faster, Roku replied to this:
```
Comment (lukaslohnicky) [inline readme.md]: > bigger 

larger?

File: readme.md [hunk 3]
```diff
@@ -44,6 +46,23 @@ end function
...
+> Could be the difference between using curlLibrary and lower-level roStreamSocket implementation. The curl library is highly optimized as we use it for all the manifest handling in streaming.
+
+The `roStreamSocket` is separated into standalone thread but still is serialized, a devices handling higher amount of data could overloaded a buffer for some moments and data could be send longer.
```
Comment (lukaslohnicky) [inline readme.md]: > a devices handling higher amount of data could overloaded a buffer for some moments and data could be send longer. 

devices handling a higher amount of data could overload a buffer for some moments and data could send for longer.

File: readme.md [hunk 4]
```diff
@@ -44,6 +46,23 @@ end function
...
+## A differences between `roUrlTransfer` and `roStreamSocket`
+
+The `roUrltransfer` is sending bigger data faster, Roku replied to this:
+> Could be the difference between using curlLibrary and lower-level roStreamSocket implementation. The curl library is highly optimized as we use it for all the manifest handling in streaming.
+
+The `roStreamSocket` is separated into standalone thread but still is serialized, a devices handling higher amount of data could overloaded a buffer for some moments and data could be send longer. 
+
+We approximately measured _230KB/s_ on **4660X - Roku Ultra** and _15KB/s_ on **3500EU - Roku Stick**.
+
+To send higher amount of data we suggest to use `roUrlTransfer` instead. Doesn't seem to be improved/fixed in near future.
```
Comment (lukaslohnicky) [inline readme.md]: > higher

larger?

File: readme.md [hunk 5]
```diff
@@ -44,6 +46,23 @@ end function
...
+
+## TLS
+
+The `roStreamSocket` is currently no supporting `TLS`, originaly there are prepared some functions to do it but not finished yet and even there is way how to encrypt data and do it on our own, the encrypting is can't be trusted by _pseudo-random_ numbers those are provided by Roku Math libs.
```
Comment (lukaslohnicky) [inline readme.md]: > no 

not

**Review by givehug — APPROVED** [review]

File: readme.md [hunk 6]
```diff
@@ -44,6 +46,23 @@ end function
...
+The `roUrltransfer` is sending larger data faster, Roku replied to this:
+> Could be the difference between using curlLibrary and lower-level roStreamSocket implementation. The curl library is highly optimized as we use it for all the manifest handling in streaming.
+
+The `roStreamSocket` is separated into standalone thread but still is serialized, devices handling a higher amount of data could overload a buffer for some moments and data could send for longer. 
+
+We approximately measured _230KB/s_ on **4660X - Roku Ultra** and _15KB/s_ on **3500EU - Roku Stick**.
+
+To send larger amount of data we suggest to use `roUrlTransfer` instead. Doesn't seem to be improved/fixed in near future.
+
+## TLS
```
Comment (lukaslohnicky) [inline readme.md]: Remove TLS section, please.

File: readme.md [hunk 7]
```diff
@@ -1,6 +1,8 @@
 # BrightWebSocket
 
-A ScreneGraph websocket client library written in BrightScript
+A ScreneGraph websocket client library written in BrightScript. It is written to work in separate task to not affect main thread of application.
```
Comment (lukaslohnicky) [inline readme.md]: > It is written to work in a separate task to not affect the main thread of the application.

File: readme.md [hunk 8]
```diff
@@ -44,6 +46,23 @@ end function
 
 For a working sample app see the "test" folder. Its contents can be zipped for
  installation as a dev channel on a Roku.
+ 
+# Additional information
+
+## A differences between `roUrlTransfer` and `roStreamSocket`
+
+The `roUrltransfer` is sending larger data faster, Roku replied to this:
```
Comment (lukaslohnicky) [inline readme.md]: > this:

this with the following:

File: readme.md [hunk 9]
```diff
@@ -44,6 +46,23 @@ end function
...
+> Could be the difference between using curlLibrary and lower-level roStreamSocket implementation. The curl library is highly optimized as we use it for all the manifest handling in streaming.
+
+The `roStreamSocket` is separated into standalone thread but still is serialized, devices handling a higher amount of data could overload a buffer for some moments and data could send for longer.
```
Comment (lukaslohnicky) [inline readme.md]: > some moments

some situations

File: readme.md [hunk 10]
```diff
@@ -44,6 +46,23 @@ end function
...
+## A differences between `roUrlTransfer` and `roStreamSocket`
+
+The `roUrltransfer` is sending larger data faster, Roku replied to this:
+> Could be the difference between using curlLibrary and lower-level roStreamSocket implementation. The curl library is highly optimized as we use it for all the manifest handling in streaming.
+
+The `roStreamSocket` is separated into standalone thread but still is serialized, devices handling a higher amount of data could overload a buffer for some moments and data could send for longer. 
+
+We approximately measured _230KB/s_ on **4660X - Roku Ultra** and _15KB/s_ on **3500EU - Roku Stick**.
+
+To send larger amount of data we suggest to use `roUrlTransfer` instead. Doesn't seem to be improved/fixed in near future.
```
Comment (lukaslohnicky) [inline readme.md]: > Doesn't seem to be

Won't be

**Pull request merged**